spec/services/media_storage/stored_path_spec.rb

```diff
@@ -41,14 +36,46 @@
     end
 
     context 'with an azure native stored path (no old aws prefixes)' do
-      let(:stored_path) { 'production/subject_location/f2eb4dbc-1353-4598-b1f6-3bf9e9a14169.jpeg' }
+      let(:stored_path) { 'subject_location/f2eb4dbc-1353-4598-b1f6-3bf9e9a14169.jpeg' }
       let(:expected_url) do
-        "#{url}/production/subject_location/f2eb4dbc-1353-4598-b1f6-3bf9e9a14169.jpeg"
+        "#{url}/subject_location/f2eb4dbc-1353-4598-b1f6-3bf9e9a14169.jpeg"
       end
 
       it 'returns the expected url' do
         expect(result).to eq(expected_url)
       end
     end
   end
+
+  describe '.media_path' do
+    let(:result) do
+      described_class.media_path(stored_path)
+    end
+
+    context 'with a migrated aws stored path' do
+      let(:stored_path) do
+        'panoptes-uploads.zooniverse.org/test/user_avatar/1e5fc9b5-86f1-4df3-986f-549f02f969a5.jpeg'
+      end
+      let(:expected_result) do
+        '/user_avatar/1e5fc9b5-86f1-4df3-986f-549f02f969a5.jpeg'
+      end
+
+      it 'returns a path without the old domain prefix and env prefix' do
+        expect(result).not_to include('panoptes-uploads.zooniverse.org/test')
+      end
+
+      it 'returns the expected path' do
+        expect(result).to eq(expected_result)
+      end
+    end
+
+    context 'with with an azure native stored path (no old aws prefixes)' do
+      let(:stored_path) { 'subject_location/f2eb4dbc-1353-4598-b1f6-3bf9e9a14169.jpeg' }
+      let(:expected_result) { 'subject_location/f2eb4dbc-1353-4598-b1f6-3bf9e9a14169.jpeg' }
```

Comment:
this might be useful if we update `stored_path` let definition
```suggestion
      let(:expected_result) { stored_path }
```